ovs/lib/storagerouter.py

```diff
@@ -114,27 +114,18 @@ def get_metadata(storagerouter_guid):
                             used_size, _ = client.run('du -B 1M -d 0 {0}'.format(storagedriver_partition.path)).split('\t')
                             directory_used_size = int(used_size)
                         except Exception as ex:
-                            logger.error('Failed to get directory usage for {0}. {1}'.format(storagedriver_partition.path, ex))
+                            logger.warning('Failed to get directory usage for {0}. {1}'.format(storagedriver_partition.path, ex))
                     used_space_by_roles += directory_used_size
 
                 partition_available_space = None
                 if disk_partition.mountpoint is not None:
-                    disk_partition_device = '/dev/{0}{1}'.format(disk.name, disk_partition.id)
+                    disk_partition_device = os.readlink(disk_partition.path).replace('../..', '/dev')
                     try:
                         output = client.run("df -Pl | grep {0}".format(disk_partition_device))
```

Comment:
I put it in a try/except because of all the conversion, line splitting, etc.
Changed the command to return required value with less parsing.